blockchain/setup.go

```diff
@@ -1,210 +1,121 @@
-		return nil, fmt.Errorf("Failed eventHub.Connect() [%s]", err)
+		return fmt.Errorf("failed to create new resource management client: %v", err)
 	}
-	setup.EventHub = eventHub
 
-	// Tell that the initialization is done
-	setup.Initialized = true
+	// Org peers join channel
+	if err = setup.admin.JoinChannel(setup.ChannelID); err != nil {
+		return fmt.Errorf("org peers failed to join the channel: %v", err)
+	}
 
-	return &setup, nil
+	fmt.Println("Initialization Successful")
+	return nil
 }
 
-// getEventHub initialize the event hub
-func getEventHub(client api.FabricClient) (api.EventHub, error) {
-	eventHub, err := events.NewEventHub(client)
-	if err != nil {
-		return nil, fmt.Errorf("Error creating new event hub: %v", err)
-	}
-	foundEventHub := false
-	peerConfig, err := client.GetConfig().GetPeersConfig()
+func (setup *FabricSetup) InstallAndInstantiateCC() error {
+
+	// Create chaincode package for our chaincode
+	ccPkg, err := packager.NewCCPackage(setup.ChaincodePath, setup.ChaincodeGoPath)
 	if err != nil {
-		return nil, fmt.Errorf("Error reading peer config: %v", err)
-	}
-	for _, p := range peerConfig {
-		if p.EventHost != "" && p.EventPort != 0 {
-			fmt.Printf("EventHub connect to peer (%s:%d)\n", p.EventHost, p.EventPort)
-			eventHub.SetPeerAddr(fmt.Sprintf("%s:%d", p.EventHost, p.EventPort),
-				p.TLS.Certificate, p.TLS.ServerHostOverride)
-			foundEventHub = true
-			break
-		}
+		return fmt.Errorf("failed to create chaincode package: %v", err)
 	}
 
-	if !foundEventHub {
-		return nil, fmt.Errorf("No EventHub configuration found")
+	// Install our chaincode on org peers
+	installCCReq := resmgmt.InstallCCRequest{Name: setup.ChainCodeID, Path: setup.ChaincodePath, Version: "1.0", Package: ccPkg}
+	_, err = setup.admin.InstallCC(installCCReq)
+	if err != nil {
+		return fmt.Errorf("failed to install cc to org peers %v", err)
 	}
 
-	return eventHub, nil
-}
+	// Set up chaincode policy
+	ccPolicy := cauthdsl.SignedByAnyMember([]string{"org1.hf.chainhero.io"})
 
-// Install and instantiate the chaincode
-func (setup *FabricSetup) InstallAndInstantiateCC() error {
-
-	// Check if chaincode ID is provided
-	// otherwise, generate a random one
-	if setup.ChaincodeId == "" {
-		setup.ChaincodeId = fcutil.GenerateRandomID()
-	}
-
-	fmt.Printf(
-		"Chaincode %s (version %s) will be installed (Go Path: %s / Chaincode Path: %s)\n",
-		setup.ChaincodeId,
-		setup.ChaincodeVersion,
-		setup.ChaincodeGoPath,
-		setup.ChaincodePath,
-	)
-
-	// Install ChainCode
-	// Package the go code and make a proposal to the network with this new chaincode
-	err := fcutil.SendInstallCC(
-		setup.Client, // The SDK client
-		setup.Channel, // The channel concerned
-		setup.ChaincodeId,
-		setup.ChaincodePath,
-		setup.ChaincodeVersion,
-		nil,
-		setup.Channel.GetPeers(), // Peers concerned by this change in the channel
-		setup.ChaincodeGoPath,
-	)
+	// Org resource manager will instantiate our chaincode on the channel
```

Comment:
Explain in more detail. Tell what is the instantiate process (regarding install process).

Reply:
@MaximeAubanel

Reply:
Done